### ISSUE TITLE
Accessibility: Unable to edit/delete date using 'Delete' or 'Backspace' button after first use

### ISSUE THREAD

**Issue by sarveshnam** [Hacker0x01/react-datepicker#1636]
**Expected behavior**
In datepicker, the date should be cleared on use of 'Delete' or 'Backspace' button.

**Actual behavior**
While using react-datepicker with redux-form, unable to clear the date after a valid date is selected / entered through textbox.

**Steps to reproduce**
1. Create a react form with redux-form and create react-datepicker field as one of the fields.
2. After loading the form, select a valid date.
3. Try to edit/delete the date using 'Delete' or 'Backspace' buttons and observe.
4. Unable to edit/delete the date after first use.

**Comment by aij** [comment]
Are you able to reproduce the problem on https://reactdatepicker.com/ ? It seems to be working as expected for me...

**Comment by DaveMitten** [comment]
Was this ever solved?

**Comment by saidas3524** [comment]
the issue is reproducible when we use the reactdatepicker along with redux-form. Is there suggested approach/way to make them work together

**Comment by Swastikyadav** [comment]
I found the issue is reproducible when we try to make DatePicker a controlled component. Instead, use it as an uncontrolled component and it will work. This means simply avoiding having `value` as an attribute in your DatePicker.

I think DatePicker has its own state management internally.

Hope this helps anyone. Thanks!

**Comment by vilasinimodem** [comment]
thank you @Swastikyadav . This solution helped me

**Comment by vilasinimodem** [comment]
But this doesnt help when you want to hold the date value and display it

**Comment by Swastikyadav** [comment]
> But this doesnt help when you want to hold the date value and display it

Sorry for not being responsive here @vilasinimodem 

I hope you have resolved this by now. But in case you haven't or anyone else may need this.
You should try to use some combination of `defaultValue`, `refs`, and `onChange handler`.
